### DISCUSSION TITLE
Equivalent Keyboard Shortcut to Create a New File in Project Panel in Zed like VS Code

### DISCUSSION THREAD

**Discussion by kresnasatya** [zed-industries/zed#9046] [Feature Requests]
Hi everyone, I have been try using Zed Editor more than weeks. So far so good. But, I wish Zed can catch up VS Code soon.

I have uncomfortable experience when create a new file (or create a file in side a folder) in Zed.

In VS Code, we can setup keybinding to create a new file inside a project in explorer panel with a keyboard shortcut. I set up in VS Code by `cmd + shift + p` then `Preferences: Open Keyboard Shortcuts (JSON)` then add a configuration below.

```
// Place your key bindings in this file to override the defaults
[
    {
        "key": "cmd+n",
        "command": "explorer.newFile"
    }
]
```

I get this keybinding shortcut from VS Code official YouTube Channel. I think Zed should create a YouTube channel too.

Short video from VS Code: https://www.youtube.com/shorts/VqOVb76IyI4

I would like to try in Zed, I try with this config below. But, it doesn't work. :(

```
[
  {
    "context": "Editor",
    "bindings": {
      "cmd-n": "project_panel::NewFile"
    }
  }
]
```

Here's a video demo when create a new file with VS Code shortcut and Zed.

https://github.com/zed-industries/zed/assets/11099186/44bdec4d-fd84-413c-a784-2c8456bf98c0

**Comment by 0x2CA** [comment]
https://github.com/zed-industries/zed/assets/19868292/99246129-a3f8-404f-999f-3e5bf360302c

**Comment by kresnasatya** [comment]
> 2024-03-08.11.01.10.mov

I'm sorry, but this is not the answer I'm looking for. I'm seeking the equivalent shortcut keybinding for create a new file like VS Code as I mention above but in Zed way.

**Comment by runofthemillgeek** [comment]
@kresnasatya I think the following hack works for the moment until there's a way to dispatch multiple actions for a given keystroke:

```json
[
  {
    "context": "Workspace",
    "bindings": {
      "cmd-alt-n": ["workspace::SendKeystrokes", "cmd-shift-e cmd-up cmd-n"]
    }
  }
]
```

**Reply by gregbarbosa** [reply]
Works for me, thank you

**Reply by Ayc0** [reply]
I did a small variation (as I wanted to insert the new file in the same folder I was):


```json
[
  {
    "context": "Workspace",
    "bindings": {
      "cmd-alt-n": ["workspace::SendKeystrokes", "cmd-shift-e cmd-n"]
    }
  }
]
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
